src/github.com/getlantern/balancer/balancer.go

```diff
@@ -102,11 +102,17 @@ func (b *Balancer) Dial(network, addr string) (net.Conn, error) {
 		trustedOnly = true
 	}
 
+	var lastDialer *dialer
 	for i := 0; i < dialAttempts; i++ {
 		d, pickErr := b.pickDialer(trustedOnly)
 		if pickErr != nil {
 			return nil, pickErr
 		}
+		if d == lastDialer {
```

Comment:
> 1. keep persistent connection to each proxy (disconnect it randomly to be less noticable).

If we can avoid getting fingerprinted here, this is also a potentially nice performance boost because it eliminates unnecessary TCP connection setup.

> always forward follow-up requests of that domain to same proxy, unless the metrics of that proxy becomes unacceptable.

I think this would solve #4902 as well.